python/desc/imsim/skyModel.py

```diff
@@ -96,15 +98,16 @@ def addNoiseAndBackground(self, image, bandpass=None, m5=None,
 
         bandPassName = self.obs_metadata.bandpass
         skyMagnitude = skyModel.returnMags()[bandPassName]
+        skySpec = skyModel.returnWaveSpec()
 
         # Since we are only producing one eimage, account for cases
         # where nsnap > 1 with an effective exposure time for the
         # visit as a whole.  TODO: Undo this change when we write
         # separate images per exposure.
-        exposureTime = photParams.nexp*photParams.exptime*u.s
+        exposureTime = photParams.nexp*photParams.exptime
 
-        skyCounts = skyCountsPerSec(surface_brightness=skyMagnitude,
-                                    filter_band=bandPassName)*exposureTime
+        # bandpass is the CatSim bandpass object
+        skyCounts = skyCountsPerSec(skySpec, bandpass, photParams)*exposureTime*u.s
```

Comment:
Instead of `skySpec`, you should just pass `skyModel`.

Reply:
OK! Is it better to pass skyModel as an argument or have skyCountsPerSec be a function of the ESOSkyModel class?